### PR TITLE
added skip config for ui build-check

### LINKED ISSUE #3016: mvn -DskipTests should skip UI tests
**Is your feature request related to a problem? Please describe.**
When spinning up the docker containers locally or during the CI, we run `sh docker/run_local_docker.sh`, which executes `mvn -DskipTests clean package`. This starts the building process of the OM server container without running backend tests, but still I believe there are selenium tests running, as I can see outputs as

```
PASS @openmetadata src/components/common/error-with-placeholder/ErrorPlaceHolder.test.tsx
```

**Describe the solution you'd like**
Is there a way we can make sure that we are able to skip them too? That could save a few minutes on unnecessary testing.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by parthp2107**
### Describe your changes :

Fix #3016

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### CODE REVIEW

File: openmetadata-ui/pom.xml [hunk 1]
```diff
@@ -137,6 +137,20 @@
...
           </execution>
+          <execution>
+            <id>yarn run build-test</id>
+            <goals>
+              <goal>yarn</goal>
+            </goals>
+            <!-- optional: the default phase is "generate-resources" -->
+            <phase>test</phase>
+            <configuration>
+              <!-- optional: if not specified, it will run gulp's default task
```
Comment (mithmatt) [inline openmetadata-ui/pom.xml:148]: Can we remove these comments?

They seem outdated

For reference on what these parameters are, one can always look at maven documentation

**Review by harshach — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/package.json [hunk 2]
```diff
@@ -83,7 +83,8 @@
     "lint": "eslint \"./src/**/*.{js,jsx,ts,tsx}\"",
     "lint:fix": "eslint './src/**/*.{js,jsx,ts,tsx}' --fix",
     "pretty": "prettier . --config './.prettierrc' --write",
-    "build-check": "yarn run lint && yarn run test",
+    "build-check": "yarn run lint",
+    "build-test": "yarn run test",
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/package.json:87]: Is this included in the GitHub CI to detect frontend unit test failures?

Comment (darth-coder00) [reply openmetadata-ui/src/main/resources/ui/package.json:91]: Yes unit tests will run for `mvn clean package`

File: openmetadata-ui/pom.xml (hunk 1 above)
Comment (parthp2107) [reply openmetadata-ui/pom.xml]: Addressed in the latest commit.

**Review by mithmatt — APPROVED** [review]

**Review by shahsank3t — APPROVED** [review]

**Comment by mithmatt** [comment]
https://github.com/open-metadata/OpenMetadata/pull/3072#issuecomment-1056683006

Does jest get installed on CI ?

Got this failure on a recent PR for  'yarn run build-test'

**Pull request merged**